packages/fetchai/skills/aries_alice/behaviours.py

```diff
@@ -0,0 +1,193 @@
+        )
+        oef_search_msg = OefSearchMessage(
+            performative=OefSearchMessage.Performative.REGISTER_SERVICE,
+            dialogue_reference=oef_search_dialogues.new_self_initiated_dialogue_reference(),
+            service_description=description,
+        )
+        oef_search_msg.counterparty = self.context.search_service_address
+        dialogue = oef_search_dialogues.update(oef_search_msg)
+        if dialogue is not None:
+            self.context.outbox.put_message(message=oef_search_msg)
+            self.context.logger.info("registering Alice service on SOEF.")
+        else:
+            self.context.logger.exception(
+                "something went wrong when registering Alice service on SOEF."
+            )
+
+    def _unregister_service(self) -> None:
+        """
+        Unregister service from the SOEF.
+
+        :return: None
+        """
+        strategy = cast(AliceStrategy, self.context.strategy)
+        description = strategy.get_unregister_service_description()
+        oef_search_dialogues = cast(
+            OefSearchDialogues, self.context.oef_search_dialogues
+        )
+        oef_search_msg = OefSearchMessage(
+            performative=OefSearchMessage.Performative.UNREGISTER_SERVICE,
+            dialogue_reference=oef_search_dialogues.new_self_initiated_dialogue_reference(),
+            service_description=description,
+        )
+        oef_search_msg.counterparty = self.context.search_service_address
+        dialogue = oef_search_dialogues.update(oef_search_msg)
+        if dialogue is not None:
+            self.context.outbox.put_message(message=oef_search_msg)
+            self.context.logger.info("unregistering service from SOEF.")
+        else:
+            self.context.logger.exception(
+                "something went wrong when unregistering Alice service on SOEF."
+            )
+
+    def _unregister_agent(self) -> None:
+        """
+        Unregister agent from the SOEF.
+
+        :return: None
+        """
+        strategy = cast(AliceStrategy, self.context.strategy)
+        description = strategy.get_location_description()
+        oef_search_dialogues = cast(
+            OefSearchDialogues, self.context.oef_search_dialogues
+        )
+        oef_search_msg = OefSearchMessage(
+            performative=OefSearchMessage.Performative.UNREGISTER_SERVICE,
+            dialogue_reference=oef_search_dialogues.new_self_initiated_dialogue_reference(),
+            service_description=description,
+        )
+        oef_search_msg.counterparty = self.context.search_service_address
+        dialogue = oef_search_dialogues.update(oef_search_msg)
+        if dialogue is not None:
+            self.context.outbox.put_message(message=oef_search_msg)
+            self.context.logger.info("unregistering agent from SOEF.")
+        else:
+            self.context.logger.exception(
+                "something went wrong when unregistering Alice on SOEF."
+            )
```

Comment:
You should not handle it; you should throw an error. An error here (i.e. no dialogue returned) means you coded wrong; so i'd just add an assertion or value error

Reply:
Done